src/extension/background-script/connectors/galoy.ts

```diff
@@ -419,8 +449,13 @@ class Galoy implements Connector {
         if (wallet === undefined) {
           throw new Error("Bad data received.");
         }
-        if (wallet.walletCurrency === "USD") {
-          throw new Error("USD currency support is not yet implemented.");
+        if (wallet.walletCurrency !== (this.config.currency || "BTC")) {
```

Comment:
targetWallet.walletCurrency is from blink apis, currently blink has one wallet id per currency, so it should always match since the connector is initialized with the right wallet id, however i think this check is worth keeping because it will make eventual changes to the blink api break the connector instead of failing silently with unexpected results.